stdlib/Serialization/src/Serialization.jl

```diff
@@ -1112,7 +1112,8 @@ function deserialize(s::AbstractSerializer, ::Type{Core.LineInfoNode})
         method = mod
         mod = Main
     end
-    return Core.LineInfoNode(mod, method, deserialize(s)::Symbol, deserialize(s)::Int, deserialize(s)::Int)
+    T = format_version(s) >= 19 ? Int32 : Int
+    return Core.LineInfoNode(mod, method, deserialize(s)::Symbol, Int32(deserialize(s)::T), Int32(deserialize(s)::T))
```

Comment:
I would just allow either type here, so deserializing works even without a header specifying the version.

Reply:
Makes sense.

Reply:
Is https://github.com/JuliaLang/julia/pull/44548/commits/bd6615ea85b5cdd60647446e88ff9261b2f512fe what you meant?

Reply:
We can add a `::Union{Int32, Int}` typeassert; the compiler will split that very nicely.